File: Benchmarks/LiteralComparison.cs
using System;
using System.Text.Patterns;
using System.Text.RegularExpressions;
using BenchmarkDotNet.Attributes;

namespace Benchmarks {
	[ClrJob, CoreJob, CoreRtJob]
	[MemoryDiagnoser]
	public class LiteralComparison {

		readonly Pattern Literal = "Hello";

		readonly Regex Regex = new Regex("^Hello");

		[Params("Hello", "World", "H", "Failure")]
		public String Source { get; set; }

		[Benchmark]
		public Result Stringier() => Literal.Consume(Source);

		[Benchmark(Baseline = true)]
		public Match MSRegex() => Regex.Match(Source);

	}
}

File: Benchmarks.FSharp/LiteralComparison.fs
namespace Benchmarks

open System.Text.Patterns
open FParsec
open BenchmarkDotNet.Attributes
open BenchmarkDotNet.Diagnosers

[<ClrJob>]
[<MemoryDiagnoser>]
type LiteralComparison() =
    let pattern = p"Hello"
    let parsec = attempt (pstringCI "Hello")

    [<Params("Hello", "World", "H")>]
    member val Source = "" with get, set

    [<Benchmark>]
    member this.PatternConsume() = pattern.Consume(this.Source)

    [<Benchmark(Baseline = true)>]
    member this.ParsecRun() = run parsec this.Source